web/src/main/java/pl/edu/agh/papaya/security/AuthenticationService.java

```diff
@@ -1,28 +1,68 @@
 package pl.edu.agh.papaya.security;
 
+import com.github.javafaker.Faker;
+import com.github.javafaker.Name;
 import java.util.Collections;
 import java.util.List;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.userdetails.User;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+import pl.edu.agh.papaya.model.User;
+import pl.edu.agh.papaya.repository.UserRepository;
 
 @Service
 public class AuthenticationService {
 
     @Autowired
     private TokenRepository tokenRepository;
 
+    @Autowired
+    private UserRepository userRepository;
+
     /**
      * Upon successful login, this method returns a generated bearer token, which shall be used as an authorization
      * method for subsequent requests.
      * <p>
      * The returned token unambiguously identifies the authenticated user and his roles.
      *
      * @return the generated token
+     *
+     * @throws pl.edu.agh.papaya.security.AuthenticationException when authentication fails
      */
-    public String logIn(String username) {
+    @Transactional(isolation = Isolation.REPEATABLE_READ)
+    public String logIn(String username) throws AuthenticationException {
+        validateEmail(username);
+
+        User user;
+        User existingUser = userRepository.findByEmail(username);
+        if (existingUser == null) {
+            Name name = new Faker().name();
+            user = new User();
+            user.setEmail(username);
+            user.setFirstName(name.firstName());
+            user.setLastName(name.lastName());
+            userRepository.save(user);
+        } else {
+            user = existingUser;
+        }
+
+        return tokenRepository.newToken(toSpringUser(user));
+    }
+
+    private void validateEmail(String email) throws AuthenticationException {
```

Comment:
Ja bym to osobiście pewnie zrobiła na zasadzie booleanowej metody wołanej wyżej

Reply:
Noo tylko wtedy tracimy potencjalnie ważny exception